graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx

```diff
@@ -0,0 +1,145 @@
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React from 'react';
+
+import { OverlayTrigger, PaginatedList, SearchForm, Spinner, Icon } from 'components/common';
+import { Row, Col, Table, Popover, Button } from 'components/bootstrap';
+import CacheTableEntry from 'components/lookup-tables/CacheTableEntry';
+import withPaginationQueryParameter from 'components/common/withPaginationQueryParameter';
+import { LookupTableCachesActions } from 'stores/lookup-tables/LookupTableCachesStore';
+import type { LookupTableCache, PaginationType } from 'logic/lookup-tables/types';
+import type { PaginationQueryParameterResult } from 'hooks/usePaginationQueryParameter';
+
+import Styles from './Overview.css';
+
+const buildHelpPopover = () => {
+  return (
+    <Popover id="search-query-help" className={Styles.popoverWide} title="Search Syntax Help">
+      <p><strong>Available search fields</strong></p>
+      <Table condensed>
+        <thead>
+          <tr>
+            <th>Field</th>
+            <th>Description</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td>id</td>
+            <td>Cache ID</td>
+          </tr>
+          <tr>
+            <td>title</td>
+            <td>The title of the cache</td>
+          </tr>
+          <tr>
+            <td>name</td>
+            <td>The reference name of the cache</td>
+          </tr>
+          <tr>
+            <td>description</td>
+            <td>The description of cache</td>
+          </tr>
+        </tbody>
+      </Table>
+      <p><strong>Examples</strong></p>
+      <p>
+        Find caches by parts of their names:<br />
+        <kbd>name:guava</kbd><br />
+        <kbd>name:gua</kbd>
+      </p>
+      <p>
+        Searching without a field name matches against the <code>title</code> field:<br />
+        <kbd>guava</kbd> <br />is the same as<br />
+        <kbd>title:guava</kbd>
+      </p>
+    </Popover>
+  );
+};
+
+type Props = {
+  caches: LookupTableCache[],
+  pagination: PaginationType,
+  paginationQueryParameter: PaginationQueryParameterResult,
+};
+
+const CachesOverview = ({ caches, pagination, paginationQueryParameter }: Props) => {
+  const { currentPage, currentPageSize, resetPage } = React.useMemo(() => ({
+    currentPage: paginationQueryParameter.page || 1,
+    currentPageSize: paginationQueryParameter.pageSize || 10,
+    resetPage: paginationQueryParameter.resetPage,
+  }), [paginationQueryParameter]);
+
+  const onPageChange = (newPage: number, newPerPage: number) => {
+    LookupTableCachesActions.searchPaginated(newPage, newPerPage, pagination.query);
+  };
+
+  const onSearch = (query: string, resetLoadingStateCb: () => void) => {
+    resetPage();
+    LookupTableCachesActions.searchPaginated(1, currentPageSize, query).then(resetLoadingStateCb);
+  };
+
+  const onReset = () => {
+    resetPage();
+    LookupTableCachesActions.searchPaginated(currentPage, currentPageSize);
+  };
```

Comment:
Let's use `useCallback` for `onPageChange `, `onSearch`, `onReset` since this component has many children?

Reply:
Similar case in to `DataAdaptersOverview` and `LookupTablesOverview`.